src/main/java/com/google/javascript/clutz/DeclarationGenerator.java

```diff
@@ -1116,15 +1117,25 @@ private void visitVarDeclaration(String name, JSType type) {
 
     private void visitTemplateTypes(ObjectType type) {
       if (type.hasAnyTemplateTypes() && !type.getTemplateTypeMap().isEmpty()) {
-        emit("<");
-        Iterator<TemplateType> it = type.getTemplateTypeMap().getTemplateKeys().iterator();
-        while (it.hasNext()) {
-          emit(it.next().getDisplayName());
-          if (it.hasNext()) {
-            emit(",");
+        // Closure compiler add a fake template types like IObject#KEY or IObject#Value
+        // for several types. The following code is a workaround waiting this bug is fixed in the
+        // closure compiler
+        // TODO(dramaix): remove this code when the bug is fixed in JsCompiler.
+        List<String> realTemplateType = new ArrayList<>();
+
+        for (TemplateType templateType: type.getTemplateTypeMap().getTemplateKeys()) {
+          String displayName = templateType.getDisplayName();
+
+          if (displayName.indexOf('#') < 0) {
```

Comment:
Good point, my latest thinking is that `IObject#` type arguments should be skipped for all but `IObject` itself.

I made some changes onto your PR and opened a main repo PR here https://github.com/angular/clutz/pull/280. Added a few more tests too. Pls, review it there.